File: tests/test_apps/php/smoke_test.php
<?php

declare(strict_types=1);

use PHPUnit\Framework\TestCase;

final class SmokeTest extends TestCase
{
    public function testPackageLoads(): void
    {
        $this->assertTrue(extension_loaded('html_to_markdown'));
    }

    public function testBasicConversion(): void
    {
        $html = '<p>Hello World</p>';
        $result = html_to_markdown_convert($html);
        $this->assertStringContainsString('Hello World', $result);
    }

    public function testWithOptions(): void
    {
        $html = '<h1>Title</h1>';
        $result = html_to_markdown_convert($html);
        $this->assertStringStartsWith('#', $result);
    }

    public function testEmptyInput(): void
    {
        $result = html_to_markdown_convert('');
        $this->assertSame('', $result);
    }
}

File: tests/test_apps/java/src/test/java/dev/kreuzberg/htmltomarkdown/SmokeTest.java
package dev.kreuzberg.htmltomarkdown;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

class SmokeTest {
    @Test
    void testPackageLoads() {
        assertNotNull(HtmlToMarkdown.class);
    }

    @Test
    void testBasicConversion() {
        String html = "<p>Hello World</p>";
        String result = HtmlToMarkdown.convert(html);
        assertTrue(result.contains("Hello World"));
    }

    @Test
    void testWithOptions() {
        String html = "<h1>Title</h1>";
        String result = HtmlToMarkdown.convert(html);
        assertTrue(result.startsWith("#"));
    }

    @Test
    void testEmptyInput() {
        String result = HtmlToMarkdown.convert("");
        assertEquals("", result);
    }
}
